File: projects/03-patient-vitals/backend/processor/transform.js
"use strict";

function toChartRow(windowMessage) {
  const windowSummary = typeof windowMessage === "string" ? JSON.parse(windowMessage) : windowMessage;
  const patientId = windowSummary.site_id || "patient-1";
  return {
    sensor_type: windowSummary.sensor_type,
    sort_key: `${windowSummary.window_end}#${patientId}`,
    window_end: windowSummary.window_end,
    window_start: windowSummary.window_start,
    site_id: patientId,
    unit: windowSummary.unit || "",
    count: windowSummary.count,
    min: windowSummary.min,
    max: windowSummary.max,
    avg: windowSummary.avg,
    latest: windowSummary.latest,
    alerts: windowSummary.alerts || [],
  };
}

module.exports = { toChartRow };

File: projects/27-river-flood-monitoring/backend/processor/transform.py
"""SQS message body to DynamoDB item; sort_key = window_end#site_id keeps the two reaches distinct per signal."""
import json


def to_item(body):
    data = json.loads(body) if isinstance(body, str) else body
    site = data.get("site_id", "reach-a")
    return {
        "sensor_type": data["sensor_type"],
        "sort_key": f"{data['window_end']}#{site}",
        "site_id": site,
        "unit": data.get("unit", ""),
        "window_start": data["window_start"],
        "window_end": data["window_end"],
        "count": data["count"],
        "min": data["min"],
        "max": data["max"],
        "avg": data["avg"],
        "latest": data["latest"],
        "rise_mph": data.get("rise_mph", 0),
        "alerts": data.get("alerts", []),
    }
